sdk/eventhub/Azure.Messaging.EventHubs/samples/Program.cs

```diff
@@ -42,27 +52,172 @@ public static async Task Main(string[] args)
             Console.WriteLine("=========================================");
             Console.WriteLine();
 
-            // Prompt for the connection string, if it wasn't passed.
+            INamedSample sample = RetrieveSample();
 
-            while (String.IsNullOrEmpty(parsedArgs.ConnectionString))
+            if (sample == null)
             {
-                Console.Write("Please provide the connection string for the Event Hubs namespace that you'd like to use and then press Enter: ");
-                parsedArgs.ConnectionString = Console.ReadLine().Trim();
+                Console.WriteLine();
+                Console.WriteLine();
+                Console.WriteLine("Quitting...");
+                Console.WriteLine();
+
+                return;
+            }
+            else
+            {
+                Console.WriteLine();
+                Console.WriteLine();
+                Console.WriteLine();
+                Console.WriteLine("-------------------------------------------------------------------------");
+                Console.WriteLine($"Running: { sample.Name }");
+                Console.WriteLine("-------------------------------------------------------------------------");
                 Console.WriteLine();
             }
 
-            // Prompt for the Event Hub name, if it wasn't passed.
+            // Run the chosen sample
 
-            while (String.IsNullOrEmpty(parsedArgs.EventHub))
+            if (sample is IEventHubsSample eventHubsSample)
             {
-                Console.Write("Please provide the name of the Event Hub that you'd like to use and then press Enter: ");
-                parsedArgs.EventHub = Console.ReadLine().Trim();
+                PromptConnectionString(parsedArgs);
+                PromptEventHubName(parsedArgs);
+
+                await eventHubsSample.RunAsync(parsedArgs.ConnectionString, parsedArgs.EventHub);
+            }
+            else if (sample is IEventHubsIdentitySample identitySample)
+            {
+                PromptFullyQualifiedNamespace(parsedArgs);
+                PromptEventHubName(parsedArgs);
+                PromptTenantId(parsedArgs);
+                PromptClientId(parsedArgs);
+                PromptSecret(parsedArgs);
+
+                await identitySample.RunAsync(parsedArgs.FullyQualifiedNamespace,
+                                              parsedArgs.EventHub,
+                                              parsedArgs.TenantId,
+                                              parsedArgs.ClientId,
+                                              parsedArgs.Secret);
+            }
+        }
+
+        /// <summary>
+        ///   It prints a list of options to console for the user. It reads the user's
+        ///   choice then it returns the chosen sample.
+        /// </summary>
+        ///
+        /// <returns>A sample or null if that could not be determined.</returns>
+        ///
+        private static INamedSample RetrieveSample()
```

Comment:
> Since this contains only two members, I think it would be better to duplicate them in the interface for each sample category to keep things localized at a glance. Each category of sample will need its own means of running the samples that would have to have direct knowledge of its dependencies.

@jsquire 

That is a very good point. This interface helps to break the ```Main``` into two sections:
- Part 1, retrieve the sample
- Part 2, run the sample

We could remove the interface and have a struct like this:

```csharp

public struct Choice
{
      public int? Choice;
      public SampleCategory? SampleCategory;
}

public enum SampleCategory
{
     EventHubs,
     Identity
}

private static Choice RetrieveSample()
{
    ...
}

```

Essentially we would remove ```INamedSample``` and we would add an enum ```SampleCategory```?

Look forward to knowing your thoughts about it.

Reply:
I love your idea, ISample is much more compact and simplifies a lot the naming of the interface.

> somehow it feels heavier than ideal either way.

What if, we introduced a Menu class like:

```csharp

using(var menu = new Menu())
{
   ISample sample = menu.RetrieveSample()

  // (... continues as it does today)
}

public class Menu: IDisposable
{
    private readonly IEnumerable<IEventHubsSample> Samples;
    private readonly IEnumerable<IEventHubsIdentitySample> IdentitySamples;

    public Menu()
    {
        Samples = LocateSamples<IEventHubsSample>();
        IdentitySamples = LocateSamples<IEventHubsIdentitySample>();
    }

    public ISample RetrieveSample()
    {
          // Here the logic we are using today
          // that prints the samples
    }
}

```

Another idea could be providing the user with arrow/color navigation? Not sure how easy that would be as I have never done it before.

Reply:
I love the idea, but I'm hesitant to put too much more engineering effort into the runner.   Given some internal chatter about "how do we make samples the same across all of these libraries", it seems like there's going to be standardization on "package each sample as a console application" in the not too distant future.   Though it's not yet an actionable standard, it does give me pause about doing too much tuning over the sample runner user experience or design as we've got it now.   

I think our best course of action is to keep things as simple and close to what we've got, even if that's a bit less elegant than would be ideal for a crafted user experience.  I'm excited to get this sample into our nightly test run, as it will be the first coverage that we have on the true identity path.   I think that is a huge win, as it having the working sample itself for folks to see in action.